libraries/DataFlash/LogFile.cpp

```diff
@@ -1799,24 +1799,36 @@ void DataFlash_Class::Log_Write_AttitudeView(AP_AHRS_View &ahrs, const Vector3f
 void DataFlash_Class::Log_Write_Current(const AP_BattMonitor &battery)
 {
     if (battery.num_instances() >= 1) {
+        float temp;
+        bool has_temp = battery.get_temperature(temp, 0);
         struct log_Current pkt = {
             LOG_PACKET_HEADER_INIT(LOG_CURRENT_MSG),
             time_us             : AP_HAL::micros64(),
             battery_voltage     : battery.voltage(0),
             current_amps        : battery.current_amps(0),
             current_total       : battery.current_total_mah(0),
+            temperature         : (int16_t)((has_temp ? temp : 0) * 100),
         };
+        for (uint8_t i = 0; i < MAVLINK_MSG_BATTERY_STATUS_FIELD_VOLTAGES_LEN; i++) {
+            pkt.cell_voltages[i] = battery.get_cell_voltages(0)[i] + 1;
```

Comment:
Yeah the 10 times thing was a bit of a hack to see if it made graphing nice, will swap it over.

The +1 was to make log review saner... Basically we internally stored voltages with the MAVLink representation of invalid values for a cell (UINT16_MAX) which makes sending the mavlink message handy, and avoids the fact that a dead cell in the middle of a pack actually would be reported as 0. However for the DF log it has the downside of really messing up scaling when you try and graph the cell voltages. Nudging the value by 1 mV means that a unknown cell shows up as 0 when graphed which doesn't destroy the ranges nearly as badly, but also that an actually dead cell would be a distinct value from an unknown cell.

As far as the MAVLINK_MSG_BATTERY_STATUS_FIELD_VOLTAGES, that is what we are trying to hang the voltages array off of for length, but when I include the GCS_MAVLink/GCS.h header in AP_BattMonitor, I get dependency problems. I think I can peak directly at the msg_battery_status header though. I'll give that a shot.